src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
+
+  return sum/men.length;
```

Comment:
***Required**
1. Looks simple and understandable but in this case, will be better to use reduce.
2. Always try to use a map instead forEach to avoid mutation. And the map works faster.
[Iterators and Generators](https://mate-academy.github.io/style-guides/javascript.html#iterators-and-generators)
```
  const menAgeSum = men.reduce((sumAges, person) => {
    return sumAges + (person.died - person.born) / menCentury.length;
  }, 0);
```